rust/flexbuffers/src/buffer.rs

```diff
@@ -0,0 +1,76 @@
+use std::ops::{Deref, Range};
+
+/// The underlying buffer that is used by a flexbuffer Reader. 
+///
+/// This allows for custom buffer implementations as long as they can be viewed as a &[u8].
+pub trait Buffer: Deref<Target = [u8]> + Sized {
+    // The `BufferString` allows for a buffer to return a custom string which will have the
+    // lifetime of the underlying buffer. A simple `std::str::from_utf8` wouldn't work since that
+    // returns a &str, which is then owned by the callee (cannot be returned from a function).
+    //
+    // Example: During deserialization a `BufferString` is returned, allowing the deserializer
+    // to "borrow" the given str - b/c there is a "lifetime" guarantee, so to speak.
+    /// A BufferString which will live at least as long as the Buffer itself.
+    ///
+    /// Must be valid UTF-8, and only generated from the `buffer_str` function Result.
```

Comment:
>  Must be valid UTF-8, and only generated from the `buffer_str` function Result.

I think `str` guarantees UTF-8 so we don't need to mention that it must be so.

Reply:
Okay, I changed the wording a little bit to still mention UTF-8, as I've seen a few developers forget that rust `str` type must be UTF-8 format and not ASCII / some other format. 

It also serves as a reminder that flexbuffers always use UTF-8 for strings (correct me if I'm off the mark there).